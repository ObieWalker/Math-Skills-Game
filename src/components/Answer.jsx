import React from "react";

const Answers = (props) => {

  return (
    <div className="col-5">
      {props.selectedNumbers.map((number, i)=>
      <span className="btn" key={i} onClick={() => props.unselectNumber(number)}>{number}</span>)}
    </div>
  )
}

export default Answers;