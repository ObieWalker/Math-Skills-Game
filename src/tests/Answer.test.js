import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import Answer from '../../src/components/Answer';

let mountedForm, props, wrapper;
const unselectNumber = jest.fn();
const selectedNumbers = [66, 7]


const getComponent = () => {
  if (!mountedForm) {
    props = {
      unselectNumber,
      selectedNumbers
    };
    mountedForm = shallow(<Answer {...props} />);
  }
  return mountedForm;
};

describe('Answers Component', () => {
  beforeEach(() => {});

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });
  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper.find('div').length).toBe(1);
    expect(wrapper.find('span').length).toBe(2);
  });
});
