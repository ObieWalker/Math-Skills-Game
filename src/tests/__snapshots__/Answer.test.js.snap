// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answers Component renders component successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Answers
    selectedNumbers={
      Array [
        66,
        7,
      ]
    }
    unselectNumber={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="btn"
          onClick={[Function]}
        >
          66
        </span>,
        <span
          className="btn"
          onClick={[Function]}
        >
          7
        </span>,
      ],
      "className": "col-5",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": 66,
          "className": "btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 66,
        "type": "span",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": 7,
          "className": "btn",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": 7,
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="btn"
            onClick={[Function]}
          >
            66
          </span>,
          <span
            className="btn"
            onClick={[Function]}
          >
            7
          </span>,
        ],
        "className": "col-5",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": 66,
            "className": "btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 66,
          "type": "span",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": 7,
            "className": "btn",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": 7,
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
